velox/exec/tests/ArrowStreamTest.cpp

```diff
@@ -0,0 +1,241 @@
+    PrivateData* private_data() {
+      return reinterpret_cast<PrivateData*>(stream_->private_data);
+    }
+    const std::shared_ptr<ArrowReader>& reader() {
+      return private_data()->reader_;
+    }
+  };
+
+  void ExportArrowStream(
+      std::shared_ptr<ArrowReader> reader,
+      struct ArrowArrayStream* out) {
+    out->get_schema = ExportedArrayStream::StaticGetSchema;
+    out->get_next = ExportedArrayStream::StaticGetNext;
+    out->get_last_error = ExportedArrayStream::StaticGetLastError;
+    out->release = ExportedArrayStream::StaticRelease;
+    out->private_data = new ExportedArrayStream::PrivateData{std::move(reader)};
+  }
+
+  void testArrowStreamOutput(
+      const std::vector<RowVectorPtr>& vectors,
+      const RowTypePtr& type) {
+    auto reader = std::make_shared<ArrowReader>(pool_, vectors, type);
+    struct ArrowArrayStream arrowStream;
+    ExportArrowStream(reader, &arrowStream);
+    auto plan = std::make_shared<core::ArrowStreamNode>(
+        "0",
+        type,
+        std::move(std::make_shared<ArrowArrayStream>(arrowStream)),
+        pool_.get());
+    assertQuery(plan, "select * from tmp");
+  }
+
+  std::shared_ptr<memory::MemoryPool> pool_{memory::getDefaultMemoryPool()};
+};
+
+TEST_F(ArrowStreamTest, basic) {
+  vector_size_t batchSize = 1000;
+  std::vector<RowVectorPtr> vectors;
+  for (int32_t i = 0; i < 3; ++i) {
+    auto c0 = makeFlatVector<bool>(
+        batchSize, [](auto row) { return (row / 10) % 2 == 0; }, nullEvery(3));
+    auto c1 = makeFlatVector<int8_t>(
+        batchSize, [](auto row) { return row % 100; }, nullEvery(5));
+    auto c2 = makeFlatVector<int16_t>(
+        batchSize, [&](auto row) { return row * i; }, nullEvery(2));
+    auto c3 = makeFlatVector<int32_t>(
+        batchSize, [](auto row) { return row; }, nullEvery(5));
+    auto c4 = makeFlatVector<int64_t>(
+        batchSize,
+        [&](vector_size_t row) { return batchSize * i + row; },
+        nullEvery(5));
+    auto c5 = makeFlatVector<float>(
+        batchSize, [](vector_size_t row) { return row * 1.1; }, nullEvery(5));
+    auto c6 = makeFlatVector<double>(
+        batchSize, [](vector_size_t row) { return row * 1.3; }, nullEvery(11));
+    auto c7 = makeFlatVector<StringView>(
+        batchSize,
+        [](vector_size_t row) { return StringView(std::to_string(row % 100)); },
+        nullEvery(7));
+    auto c8 = makeFlatVector<StringView>(
+        batchSize,
+        [](vector_size_t row) { return StringView(std::to_string(row % 100)); },
+        nullEvery(7));
+    vectors.push_back(makeRowVector({c0, c1, c2, c3, c4, c5, c6, c7, c8}));
+  }
+  createDuckDbTable(vectors);
+
+  auto type = ROW({
+      {"c0", BOOLEAN()},
+      {"c1", TINYINT()},
+      {"c2", SMALLINT()},
+      {"c3", INTEGER()},
+      {"c4", BIGINT()},
+      {"c5", REAL()},
+      {"c6", DOUBLE()},
+      {"c7", VARCHAR()},
+      {"c8", VARBINARY()},
+  });
+
+  testArrowStreamOutput(vectors, type);
```

Comment:
This method is used only once. Let's remove it and put the code in here directly.

Reply:
Fixed.